### DISCUSSION TITLE
Fields from current record in slash commands?

### DISCUSSION THREAD

**Discussion by RickSeiden** [arnoudkooi/SN-Utils#438] [Q&A] [answered]
Is it possible to add a field from the current record to a slash command.

For example, if I'm on an update set in dev, and I want to jump to the same update set in test's retrieved update sets, I could go to this URL:

https://mytestinst.service-now.com/sys_remote_update_set_list.do?sysparm_query=update_source%3D<sys_id of source>%5Ename%3D<update set name>

If I want to build a slash command to do that, I could hard code everything up to "name%3D" in the slash command, but how can I have the slash command pull the name field from the current record?  Is that even possible?

I know there my be other ways to solve this specific use case.  I'm more interested in the generic use case where I want to put any field on a form into a slash command in the same way I would put $sys_id or $table_name.

Thanks!

**Comment by arnoudkooi** [comment]
Interesting idea, SN Devs Slackers like it as well :)
I'll give it a thought.
Do you see other use cases as well?

**Comment by RickSeiden** [comment]
In the years I've been using this, it's the first time I've come across a
use case, so I can't say that I have any others. I thought maybe there was
a way to do it already. If there's not, I'm not sure my single use case
warrants a code change, to be honest.

**Comment by tomasz-q** [comment]
You can build your slash command using JavaScript and simply use g_form.getValue('...').

**Reply by arnoudkooi** [reply]
Great suggestion :)

**Comment by RickSeiden** [comment]
You can what now? How do you build a slash command with JavaScript?!?!

**Reply by tomasz-q** [reply, marked as answer]
With javascript: prefix. Here is an example. I use it to open a new update set form while being on the rm_story form. The update set's name is generated based on the current story.
```
javascript:(function () {
    var initials = 'NN';
    var g_user = window.g_user || window.frames[0].g_user;
    var g_form = window.g_form || window.frames[0].g_form;
    
    if (g_form.tableName != 'rm_story') {
        g_form.addInfoMessage('This slash command works on rm_story form only!');
        return
    }
    
    if (g_user) {
      initials = g_user.firstName[0] + g_user.lastName[0];   
    }

    var number = g_form.getValue('number');
    var name = initials + ' - ' + number + ' - ' + g_form.getValue('short_description');
    var frame = top.gsft_main;
    if (!frame) {
        frame = top;
    }

    frame.location = '/sys_update_set.do?sys_id=-1&sys_target=sys_update_set&sysparm_query=name='+name;
})() 
```

**Reply by vedang444** [reply]
Holy moly!!! Building slash commands with javascript! Are you kidding me right now @arnoudkooi !!
Thanks a ton @RickSeiden . I feel like I've been sleeping under a rock this whole time.

Going on it immediately 🤩🤩

**Reply by codaroma** [reply]
btw this can also be done with a bookmarklet.

**Comment by RickSeiden** [comment]
And that's the slash command, instead of something like
"/$table_name.do?sysparm_query=sys_id=$sys_id" ??

**Reply by arnoudkooi** [reply]
I did a simple test (that makes no sense) but this does seem to work...

```javascript
javascript:window.open("https://www.google.com/search?q=" + (g_form?.getValue('number') || ""))
```

**Comment by RickSeiden** [comment]
That is friggin' AMAZING!  I'm going to share that at work tomorrow, and I
bet people are going to love it!

THANK YOU SO MUCH!

**Comment by arnoudkooi** [comment]
Cool, did you get it to work? If so, please share the command you ended up using.

**Comment by RickSeiden** [comment]
I did!  Here it is:

javascript:window.open("https://<instance>.service-now.com/sys_remote_update_set_list.do?sysparm_query=update_source%3D<sys_id>%5Ename%3D"+g_form.getValue("name"));

Works like a charm!

Thanks again, @tomasz-q, and thanks again for a great tool, @arnoudkooi!
